FirebaseSessions/Sources/ApplicationInfo.swift

```diff
@@ -63,4 +66,15 @@ class ApplicationInfo: ApplicationInfoProtocol {
   var mccMNC: String {
     return FIRSESValidateMccMnc(networkInfo.mobileCountryCode, networkInfo.mobileNetworkCode) ?? ""
   }
+
+  var environment: String {
+    if let env = ProcessInfo.processInfo.environment["FIREBASE_RUN_ENVIRONMENT"] {
+      // If the variable is set without any value, assume it is PROD.
+      if (env.trimmingCharacters(in: .whitespaces).isEmpty) {
+        return "PROD"
+      }
+      return env
+    }
+    return "PROD"
```

Comment:
I think we can use an enum here to increase clarity:

```
    enum DevEnvironment: String {
        case prod   = "prod"
        case staging   = "staging"
        case autopush = "autopush"
    }
```

The `DevEnvironment(rawValue: String)` constructor will return a `DevEnvironment?`, so we can construct simply, and use a default value in case there's no match:

```
  let env = ProcessInfo...
  return DevEnvironment(rawValue: env.lowercased()) ?? DevEnvironment.prod
```


I think this will also ensure if we add or remove an env, switch statements will complain that we haven't exhausted all possibilities, like in `convertLogEnvironment`.